.github/workflows/code-formatting.yml

```diff
@@ -71,13 +73,17 @@ jobs:
             commit: --signoff
 
       - name: pylint
-        if: ${{ !contains( github.event.pull_request.labels.*.name, 'skip-docs') }}
+        if: ${{ steps.changed-files.outputs.any_changed == 'true' && !contains( github.event.pull_request.labels.*.name, 'skip-docs') }}
+        env:
+          # only *.py files included
+          CHANGED_FILES: "${{ steps.changed-files.outputs.all_changed_files }}"
         run: |
           pip install pylint
-          
-          CHANGED_FILES=$(git diff --name-only --diff-filter=d --merge-base origin/main nemo/ | grep '\.py$' || true)
```

Comment:
We should check against PR target, otherwise there can be failures. "${{ steps.changed-files.outputs.all_changed_files }}" contains all changed "*.py" files